kernel/kernel-api/src/test/scala/io/delta/kernel/internal/files/ParsedLogDataSuite.scala

```diff
@@ -0,0 +1,378 @@
+    override def getSchema: StructType = null
+    override def getColumnVector(ordinal: Int): ColumnVector = null
+    override def getSize: Int = 0
+  }
+
+  /////////////
+  // General //
+  /////////////
+
+  test("Throws on unknown log file") {
+    val fileStatus = FileStatus.of("unknown", 0, 0)
+    val exMsg = intercept[IllegalArgumentException] {
+      ParsedLogData.forFileStatus(fileStatus)
+    }.getMessage
+    assert(exMsg.contains("Unknown log file type"))
+  }
+
+  test("Throws on version < 0") {
+    val exMsg = intercept[IllegalArgumentException] {
+      ParsedLogData.forInlineData(-1, ParsedLogType.RATIFIED_INLINE_COMMIT, emptyColumnarBatch)
+    }.getMessage
+    assert(exMsg.contains("version must be non-negative"))
+  }
+
+  test("Throws on both fileStatusOpt and inlineDataOpt present") {
+    val fileStatusOpt = Optional.of(deltaFileStatus(5))
+    val inlineDataOpt = Optional.of(emptyColumnarBatch)
+    val exMsg = intercept[IllegalArgumentException] {
+      new ParsedLogData(10, ParsedLogType.PUBLISHED_DELTA, fileStatusOpt, inlineDataOpt)
+    }.getMessage
+    assert(exMsg.contains("Exactly one of fileStatusOpt or inlineDataOpt must be present"))
+  }
+
+  test("Throws on both fileStatusOpt and inlineDataOpt empty") {
+    val exMsg = intercept[IllegalArgumentException] {
+      new ParsedLogData(10, ParsedLogType.PUBLISHED_DELTA, Optional.empty(), Optional.empty())
+    }.getMessage
+    assert(exMsg.contains("Exactly one of fileStatusOpt or inlineDataOpt must be present"))
+  }
+
+  ////////////
+  // Deltas //
+  ////////////
+
+  test("Correctly parses published delta file") {
+    val fileStatus = deltaFileStatus(5)
+    val parsed = ParsedLogData.forFileStatus(fileStatus)
+
+    assert(parsed.version == 5)
+    assert(parsed.`type` == ParsedLogType.PUBLISHED_DELTA)
+    assert(parsed.getCategory == ParsedLogCategory.DELTA)
+    assert(parsed.isMaterialized)
+    assert(!parsed.isInline)
+    assert(parsed.getFileStatus == fileStatus)
+  }
+
+  test("Correctly parses staged commit file") {
+    val fileStatus = stagedCommitFile(5)
+    val parsed = ParsedLogData.forFileStatus(fileStatus)
+
+    assert(parsed.version == 5)
+    assert(parsed.`type` == ParsedLogType.STAGED_COMMIT)
+    assert(parsed.getCategory == ParsedLogCategory.DELTA)
+    assert(parsed.isMaterialized)
+    assert(!parsed.isInline)
+    assert(parsed.getFileStatus == fileStatus)
+  }
+
+  test("Delta file equality") {
+    val fileStatus1 = deltaFileStatus(5)
+    val fileStatus2 = deltaFileStatus(5)
+    val fileStatus3 = deltaFileStatus(6)
+
+    val delta1 = ParsedLogData.forFileStatus(fileStatus1)
+    val delta2 = ParsedLogData.forFileStatus(fileStatus2)
+    val delta3 = ParsedLogData.forFileStatus(fileStatus3)
+
+    assert(delta1 == delta1)
+    assert(delta1 == delta2)
+    assert(delta1 != delta3)
+  }
+
+  /////////////////
+  // Checkpoints //
+  /////////////////
+
+  test("Inline classic checkpoint") {
```

Comment:
what exactly do these two tests check?

Reply:
I updated the test names to be more helpful.

We want to test that we can create the types we want using the factory methods we want, and that we cannot create certain types using other certain factory methods